Description: prevent xfpm_battery_notify_idle() being called on a battery that has been removed
Origin: other, https://bugzilla.xfce.org/attachment.cgi?id=3912
Bug: https://bugzilla.xfce.org/show_bug.cgi?id=7851
---
 src/xfpm-battery.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/src/xfpm-battery.c
+++ b/src/xfpm-battery.c
@@ -763,6 +763,8 @@
 
     battery = XFPM_BATTERY (object);
     
+    g_idle_remove_by_data (battery);
+
     g_free (battery->priv->icon_prefix);
     
     dbus_g_proxy_disconnect_signal (battery->priv->proxy, "Changed",
